services/brig/src/Brig/API.hs

```diff
@@ -153,15 +169,24 @@ sitemap o = do
   post "/i/clients" (continue internalListClientsH) $
     accept "application" "json"
       .&. jsonRequest @UserSet
+  -- This endpoint can lead to the following events being sent to clients:
+  -- - ClientAdded event to the user
+  -- - ClientRemoved event to the user, if removing old clients due to max number of clients
+  -- - UserLegalHoldEnabled event to contacts of the user, if client type is legalhold
```

Comment:
So; other users are not informed by th clients added and removed, correct?

I wonder if it's harmful if we do do that?  Then we do not need this lazy discovery mechanism when sending

Reply:
Probably not massively harmful, but could be a performance issue. It would require an event to be sent to every user in every conversation we are part of. By doing it lazily, we spread out that cost and it also only needs to be payed by people actually sending a message (which will also get all added clients at once). Seems like this is often a good tradeoff, especially when there are just a few clients sending messages, but many passive ones.

Also, we need to provide a way for querying this information anyways, e.g. for new clients, after missing messages, or just due to unfortunate timing.